src/components/PPRoute.js

```diff
@@ -0,0 +1,16 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import React from 'react';
+import { Route } from 'react-router-dom';
+import { connect } from 'react-redux';
+import LoadingComponent from './LoadingComponent';
+
+const PPRoute = ({ isLoadingUserData, isLoadingAnnouncements, ...rest }) => {
+  const loading = isLoadingAnnouncements || isLoadingUserData;
+  return loading ? <LoadingComponent /> : <Route {...rest} />;
```

Comment:
nit: i found it easier to read if you just did 
```suggestion
  return isLoadingAnnouncements || isLoadingUserData; ? <LoadingComponent /> : <Route {...rest} />;
```